include/libkrun.h

```diff
@@ -265,7 +265,107 @@ int32_t krun_add_virtiofs2(uint32_t ctx_id,
+ *              network proxy is listening.
+ *  "fd"      - a file descriptor for an already open unixstream
+ *              connection to the userspace network proxy.
+ *  "c_mac"   - MAC address as an array of 6 uint8_t entries.
+ *  "flags"   - generic flags for the network interface.
+ *  "vflags"  - virtio-net flags for the network interface.
+ *
+ * Notes:
+ * The arguments "c_path" and "fd" are mutually exclusive. If using
+ * "fd" to point to the endpoint, the socket must be already
+ * initialized and configured as the userspace network proxy
+ * requires.
+ * If no network devices are added, networking uses the TSI backend.
+ * This function should be called before krun_set_port_map.
+ *
+ * Returns:
+ *  Zero on success or a negative error number on failure.
+ */
+int32_t krun_add_net_unixstream(uint32_t ctx_id,
+                                const char *c_path,
+                                int fd,
+                                uint8_t *const c_mac,
+                                uint32_t flags,
+                                uint32_t vflags);
+
+/**
+ * Adds an independent virtio-net device with a unixgram-based
+ * backend, such as gvproxy or vmnet-helper.
+ *
+ * The "krun_add_net_*" functions can be called multiple times for
+ * adding multiple virtio-net devices. In the guest the interfaces
+ * will appear in the same order as they are added (that is, the
+ * first added interface will be "eth0", the second "eth1"...)
+ *
+ * If no network interface is added, libkrun will automatically
+ * enable the TSI backend.
+ *
+ * Arguments:
+ *  "ctx_id"  - the configuration context ID.
+ *  "c_path"  - a null-terminated string representing the path
+ *              for the unixstream socket where the userspace
+ *              network proxy is listening.
+ *  "fd"      - a file descriptor for an already open unixstream
+ *              connection to the userspace network proxy.
+ *  "c_mac"   - MAC address as an array of 6 uint8_t entries.
+ *  "flags"   - generic flags for the network interface.
+ *  "vflags"  - virtio-net flags for the network interface.
+ *
+ * Notes:
+ * The arguments "c_path" and "fd" are mutually exclusive. If using
+ * "fd" to point to the endpoint, the socket must be already
+ * initialized and configured as the userspace network proxy
+ * requires.
+ * If no network devices are added, networking uses the TSI backend.
+ * This function should be called before krun_set_port_map.
+ * If using gvproxy in vfkit mode, NET_FLAG_VFKIT must be passed in
+ * "flags" when using "c_path" to indicate the connection endpoint.
+ *
+ * Returns:
+ *  Zero on success or a negative error number on failure.
+ */
+int32_t krun_add_net_unixgram(uint32_t ctx_id,
+                              const char *c_path,
+                              int fd,
+                              uint8_t *const c_mac,
+                              uint32_t flags,
+                              uint32_t vflags);
+
```

Comment:
Same issues as previous function.

Nice that the function are the same, if learn to use one you know how to use the other.

Reply:
Also fixed for this one.